internal/provider/resource_password_test.go

```diff
@@ -559,13 +590,17 @@ func TestAccResourcePassword_StateUpgradeV0toV3(t *testing.T) {
 			configDuringUpgrade: `resource "random_password" "default" {
 						length = 12
 					}`,
-			beforeStateUpgrade: []resource.TestCheckFunc{
-				resource.TestCheckResourceAttr("random_password.default", "number", "true"),
-				resource.TestCheckNoResourceAttr("random_password.default", "numeric"),
+			beforeUpgradeStateChecks: []statecheck.StateCheck{
+				statecheck.ExpectKnownValue("random_password.default", tfjsonpath.New("number"), knownvalue.Bool(true)),
+				randomtest.ExpectNoAttribute("random_password.default", tfjsonpath.New("numeric")),
```

Comment:
A fun (and expected) quirk of our new state checks are that they will recognize if an attribute doesn't exist and fail, so certain usages of `resource.TestCheckNoResourceAttr` will not be able to be replaced by `statecheck.ExpectKnownValue` with a `knownvalue.Null()`